cloud/ec2_util.go

```diff
@@ -317,6 +317,9 @@ func cacheHostData(ctx context.Context, h *host.Host, instance *ec2.Instance, cl
 	if instance.PublicDnsName == nil {
 		return errors.New("instance missing public dns name")
 	}
+	if instance.PublicIpAddress == nil {
```

Comment:
Have you verified that this is definitely a required field? Do ipv6 hosts also have this field or will it be empty in that case?

Reply:
`PublicIpAddress -> (string)

The public IPv4 address, or the Carrier IP address assigned to the instance, if applicable.

A Carrier IP address only applies to an instance launched in a subnet associated with a Wavelength Zone.`

`Ipv6Addresses -> (list)

One or more IPv6 addresses associated with the network interface.

(structure)

Describes an IPv6 address.

Ipv6Address -> (string)

The IPv6 address.`

The `PublicIpAddress` should always have the associated ipv4 but the documentation says "one or more" ipv6 will be there but one of the examples show that that's not always the case

Reply:
Classic EC2 documentation. Could you link this?

Reply:
https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/describe-instances.html